Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -2177,17 +2198,17 @@ private JSONObject convertParamsStringToDictionary(String paramString) {
             }
         }
     }
-    
+
     private void processNextQueueItem() {
         try {
             serverSema_.acquire();
             if (networkCount_ == 0 && requestQueue_.getSize() > 0) {
                 networkCount_ = 1;
                 ServerRequest req = requestQueue_.peek();
-                
+
                 serverSema_.release();
                 if (req != null) {
-                    if (!req.isWaitingOnProcessToFinish()) {
+                    if (!req.isWaitingOnProcessToFinish() || isForceSessionEnabled()) {
```

Comment:
Reading through the code a bit deeper, there is a risk to bypassing the locks in that they are currently used to make sure:
1. We fetch GAID
2. We wait for a pending intent to be received by the activity (I think this is the actual blocking one)
3. We wait for Google Play referrer.

You can see all of these wait locks here:
https://github.com/BranchMetrics/android-branch-deep-linking/blob/master/Branch-SDK/src/io/branch/referral/Branch.java#L2342

In the forced setting, we still need GAID and we still need to wait for the Google Play referrer so if you bypass all of them, you likely will cause the initialization to be malformed (missing GAID) or missing key attribution info from the broadcast.

Reply:
I believe what you actually want to do is just bypass waitlock #2 but actually wait on the others to complete. I think you could achieve this by calling `removeProcessWaitLock` in the case where it's forced. https://github.com/BranchMetrics/android-branch-deep-linking/blob/master/Branch-SDK/src/io/branch/referral/ServerRequest.java#L553.

Or, even cleaner, just never add it if the forced boolean is set: https://github.com/BranchMetrics/android-branch-deep-linking/blob/master/Branch-SDK/src/io/branch/referral/Branch.java#L2349